lxd/instance_post.go

```diff
@@ -381,6 +381,10 @@ func instancePost(d *Daemon, r *http.Request) response.Response {
 			return operations.OperationResponse(op)
 		}
 
+		if instanceType == instancetype.VM && req.Live && shared.IsFalseOrEmpty(inst.ExpandedConfig()["migration.stateful"]) {
```

Comment:
Unfortunately this is moving instance specific (VM) logic out of the instance driver and into the generic API handler, which isn't something we want to go.

Have you identified why the error returned from qemu.MigrateSend isn't propagated to the user?

Reply:
Thanks, I agree. It works for this specific case but is a bit of a bandaid solution :)

I've pushed a better fix that utilizes `ControlSend` to return the proper error message when `migration.stateful` is empty/false for live migration. We also now set up the necessary migration connections before making the check. It seems that killing `MigrateSend` before setting these connections up causes issues with error handling/propagation.